src/client/connection.rs

```diff
@@ -176,15 +176,38 @@ impl<S: AsyncRead + AsyncWrite + Unpin + Send> Connection<S> {
                 split_payload.len() + HEADER_BYTES,
             );
 
-            let packet = Packet::new(header, split_payload);
-            self.transport.send(packet).await?;
+            self.write_to_wire(header, split_payload).await?;
         }
 
-        self.transport.flush().await?;
+        self.flush_sink().await?;
 
         Ok(())
     }
 
+    /// Sends a packet of data to the database.
+    ///
+    /// # Warning
+    ///
+    /// Please be sure the packet size doesn't exceed the largest allowed size
+    /// dictaded by the server.
+    pub async fn write_to_wire(
+        &mut self,
+        header: PacketHeader,
+        data: BytesMut,
+    ) -> crate::Result<()> {
+        self.flushed = false;
+
+        let packet = Packet::new(header, data);
+        self.transport.send(packet).await?;
+
+        Ok(())
+    }
+
+    /// Sends all pending packages to the wire.
+    pub async fn flush_sink(&mut self) -> crate::Result<()> {
```

Comment:
```suggestion
    pub(crate) async fn flush_sink(&mut self) -> crate::Result<()> {
```

Reply:
we should maybe always have this, even when connection is never public...

Reply:
taken